tests/integration/security/sds_egress/sds_egress_test.go

```diff
@@ -0,0 +1,80 @@
+package sds_egress
+
+import (
+	"context"
+	"istio.io/istio/pkg/test/echo/common/response"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/pkg/log"
+	"net/http"
+	"testing"
+	"time"
+
+	epb "istio.io/istio/pkg/test/echo/proto"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/tests/integration/security/util"
+)
+
+const (
+	// should be templated into YAML, and made interchangeable with other sites
+	externalUrl = "http://edition.cnn.com/politics"
+)
+
+// TestSdsEgressGatewayIstioMutual brings up an SDS enabled cluster and deploys an egress gateway
+// with TLS mode ISTIO_MUTUAL, and ensures that traffic from the cluster to the egress succeeds and is secured
+func TestSdsEgressGatewayIstioMutual(t *testing.T) {
```

Comment:
@mandarjog @nmittler any advice on how I can idiomatically make this test

1. Ensure sure that a request from a pod was routed to the `egress-gateway`
2. Ensure sure that this request was encrypted w/ workload certs
3. Avoid actually calling out to an external URL from the `egress-gateway` (@incfly suggested looking into Envoy's [DirectResponseAction](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route.proto#route-directresponseaction)

I don't have much experience with the test framework, if there any similar examples that would be awesome ([this](https://github.com/istio/istio/blob/master/tests/integration/pilot/outboundtrafficpolicy/helper.go) is the closest I could find wrt external requests)

Reply:
wrt using Prom metrics to figure out whether the call was over `mTLS` - I was running into [what seems to be this issue](https://discuss.istio.io/t/unknown-status-on-connection-security-policy-with-prometheus/1362), where even after forcing the cluster to use strict mTLS, the `content_security_policy` shows up as unknown. I'll keep looking into this though, tyvm for your suggestions